components/seldon-request-logger/app/log_mapping.py

```diff
@@ -105,9 +121,73 @@ def get_field_mapping(metadata):
             props = update_props_element(props, elm)
     return None if not props else {"properties": props}
 
+def init_api():
+    config = Configuration()
+    config.host = deploy_api_host
+    config.oidc_client_id = oidc_client_id
+    config.oidc_server = oidc_server
+    config.username = oidc_username
+    config.password = oidc_password
+    config.oidc_client_secret = oidc_client_secret
+    config.auth_method = oidc_auth_method
+
+    if not config.auth_method or config.auth_method is None:
+        config.auth_method = 'password_grant'
+
+    if not config.host or config.host is None:
+        print('No DEPLOY_API_HOST - will not look up metadata from Deploy')
+        return
+
+    if not config.oidc_server or config.oidc_server is None:
+        print('No OIDC_PROVIDER - auth will not be used in connecting to metadata')
+        return
+
+    auth = None
+    if config.oidc_server:
+        auth = OIDCAuthenticator(config)
+        config.access_token = auth.authenticate()
+
+    api_client = ApiClient(configuration=config, authenticator=auth)
+
+    env_api = EnvironmentApi(api_client)
+    print('connected to deploy')
+    print(env_api.read_user())
+    global metadata_api
+    metadata_api = ModelMetadataServiceApi(api_client)
+
 
-def fetch_metadata(namespace, serving_engine, inferenceservice_name):
-    # TODO: Fetch real metadata
+def fetch_user():
+    user = env_api.read_user()
+    return user
+
+
+def fetch_metadata(namespace, serving_engine, inferenceservice_name, predictor_name):
+
+    if metadata_api is None:
+        print('metadata service not configured')
+        return None
+
+    # TODO: in next iteration will only need one lookup straight to model metadata
+    # was expcting to set deployment_type=serving_engine but deployment_type does not seem to be a param
+    runtime_metadata = metadata_api.model_metadata_service_list_runtime_metadata_for_model(
+        deployment_name=inferenceservice_name,deployment_namespace=namespace,predictor_name=predictor_name)
+    if runtime_metadata is not None and runtime_metadata.runtime_metadata is not None:
+        print(runtime_metadata.runtime_metadata)
+        if len(runtime_metadata.runtime_metadata) == 0:
+            print('no runtime metadata for '+namespace+'/'+inferenceservice_name)
+            return None
+        model_uri = runtime_metadata.runtime_metadata[0].model_uri
```

Comment:
Here selecting the first model is a key logical piece that implies this will only work with predictors with one model otherwise might give undesired results using wrong model-level metadata. There should be a check for this.

Reply:
Good point, just pushed a change for this
https://github.com/SeldonIO/seldon-core/pull/3183/commits/9a9a0c86f5f875c5e48f126d26a80733fead5c0b

Reply:
Oh okay, maybe I wasn't clear. I meant if the number of models is greater than 1 then the function should return None. 
So in this case it doesn't make sense to use the first model's metadata as it would be considerably different from the predictor level metadata which should create the elastic index mapping. So use the metadata only if the number of models is exactly equal to one.